src/FakeItEasy/Core/FakeManager.cs

```diff
@@ -142,7 +143,10 @@ public virtual void RemoveRule(IFakeObjectCallRule rule)
             lock (this.allUserRules)
             {
                 var ruleToRemove = this.allUserRules.FirstOrDefault(x => x.Rule.Equals(rule));
-                this.allUserRules.Remove(ruleToRemove);
+                if (ruleToRemove is object)
+                {
+                    this.allUserRules.Remove(ruleToRemove);
+                }

```

Comment:
Remove doesn't expect a null (although I suspect it wouldn't have any ill effect)